var/spack/repos/builtin/packages/sherpa/package.py

```diff
@@ -0,0 +1,134 @@
+            description='Use the specified C++ standard when building')
+
+    variant('analysis',   default=True, description='Enable analysis components')
+    variant('mpi',        default=False, description='Enable MPI')
+    variant('python',     default=False, description='Enable Python API')
+    variant('hepmc2',     default=True, description='Enable HepMC (version 2.x) support')
+    variant('hepmc3',     default=True, description='Enable HepMC (version 3.x) support')
+    variant('hepmc3root', default=False, description='Enable HepMC (version 3.1+) ROOT support')
+    variant('rivet',      default=False, description='Enable Rivet support')
+    variant('fastjet',    default=True, description='Enable FASTJET')
+    variant('openloops',  default=False, description='Enable OpenLoops')
+    variant('recola',     default=False, description='Enable Recola')
+    variant('lhole',      default=False, description='Enable Les Houches One-Loop Generator interface')
+    variant('root',       default=False, description='Enable ROOT support')
+    variant('lhapdf',     default=True, description='Enable LHAPDF support')
+    variant('gzip',       default=False, description='Enable gzip support')
+    variant('pythia',     default=True, description='Enable fragmentation/decay interface to Pythia')
+
+    # Note that the delphes integration seems utterly broken: https://sherpa.hepforge.org/trac/ticket/305
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
+    depends_on('sqlite')
+
+    depends_on('mpi',       when='+mpi')
+    depends_on('python',    when='+python')
+    depends_on('swig',      when='+python', type='build')
+    depends_on('hepmc',     when='+hepmc2')
+    depends_on('hepmc3',    when='+hepmc3')
+    depends_on('hepmc3 +rootio', when='+hepmc3root')
+    depends_on('rivet',     when='+rivet')
+    depends_on('fastjet',   when='+fastjet')
+    depends_on('openloops', when='+openloops')
+    depends_on('recola',    when='+recola')
+    depends_on('root',      when='+root')
+    depends_on('lhapdf',    when='+lhapdf')
+    depends_on('gzip',      when='+gzip')
+    depends_on('pythia6',   when='+pythia')
+
+    for std in _cxxstd_values:
+        depends_on('root cxxstd=' + std, when='+root cxxstd=' + std)
+
+    def patch(self):
+        filter_file(r'#include <sys/sysctl.h>',
+                    '#ifdef ARCH_DARWIN\n#include <sys/sysctl.h>\n#endif',
+                    'ATOOLS/Org/Run_Parameter.C')
+
+    def configure_args(self):
+        args = []
+        args.append('--enable-shared')
+        args.append('--enable-binreloc')
+        args.append('--enable-static')
+        args.append('--enable-hepevtsize=200000')
+        args.append('--disable-manual')
+        args.append('--with-sqlite3=' + self.spec['sqlite'].prefix)
+        if self.spec.satisfies('+mpi'):
+            args.append('--enable-mpi')
```

Comment:
Can we shorten this recipe by using:
```suggestion
        args.extend(self.enable_or_disable('mpi'))
```
for variants that match the option to be enabled / disabled?

Reply:
Yes, that's nicer indeed, thanks!